superset/views/core.py

```diff
@@ -886,13 +885,6 @@ def fetch_datasource_metadata(self) -> FlaskResponse:
         datasource.raise_for_access()
         return json_success(json.dumps(sanitize_datasource_data(datasource.data)))
 
-    @app.errorhandler(500)
```

Comment:
This errorhandler is lonely here in core.py, all them other ondes are in base.py. Also ideally should have logic to return json depending on mimetype